internal/scheduler/submitcheck.go

```diff
@@ -87,6 +87,8 @@ func NewSubmitChecker(
 	}
 }
 
+var stringInterner = stringinterner.New(1000)
```

Comment:
Should this only be 1000?

Reply:
it shouldn't matter too much as it's an LRU so it'll cache the last 10k strings.  We're not really getting anything from the string interner here as these strings are already in memory and only live for the lifetime of the request (which is hopefully measure in millis)- I needed it purely because the interface desires it.

Happy to make it 10K or 100K though if you prefer.